Reneiski_Homework1/ReneiskiSelenium11/PageObjects/Pages/CheckBoxPage.cs

```diff
@@ -0,0 +1,21 @@
+﻿using OpenQA.Selenium;
+using ReneiskiSelenium11.Common.WebElements;
+
+namespace ReneiskiSelenium11.PageObjects.Pages
+{
+    public class CheckBoxPage : BasePage
+    {
+        private MyWebElement TextOfResult = new MyWebElement(By.Id("result"));
+        private MyWebElement CheckBoxHome = new MyWebElement(By.XPath("//*[@class='rct-icon rct-icon-uncheck']"));
+        private MyWebElement CheckBoxAttribute = new MyWebElement(By.XPath("//*[@class='rct-icon rct-icon-check']"));
+        private MyWebElement CheckBoxPagePath = new MyWebElement(By.XPath("//*[contains(text(), 'Check Box')]"));
+
+        public void ClickHomeCheckBox() => CheckBoxHome.Click();
+
+        public bool IsHomeCheckBoxChecked() => CheckBoxAttribute.GetAttribute("class").Contains("check");
+
+        public string GetTextResult() => TextOfResult.Text;
+
+        public void GoToCheckBoxPage() => CheckBoxPagePath.Click();
```

Comment:
also, for me it's confusing that you can call GoToCheckBoxPage from CheckBoxPage
there's a better solution here, you can do the following:
1. Create another page, call it for example MainPage, it will inherit from BasePage
2. MainPage will be a common page object for all pages opened after start page (that is on https://demoqa.com/ url)
3. Then you create a new class called LeftMenuNavigationPanel that will be a representation of this panel (see screenshot) - for this page object create a separate folder in PageObjects folder and call it "Modules"
![image](https://github.com/agrren/Reneiski_homeWork/assets/55654931/61a8fc7b-4574-4443-8bb5-c4be998ac566)
4. Then in this module you can store different elements - "Buttons" button in left menu, "Checkboxes" button in left menu etc and you can create methods inside this module that will open any of this page
5. Last step - create instance of this module in MainPage, create method like OpenCheckboxesPage and it your page objects inherit from this MainPage
6. Really last step - create in GenericPages method that will call MainPage
after all this done you will be able to call in test mehtod like GenericPages.MainPage.LeftNavigationPanel.OpenCheckboxesPage and get instance of Checkboxes page
from framework architecture side it will be more correct

Reply:
Fixed